internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/preprocessor/PreProcessor.java

```diff
@@ -0,0 +1,171 @@
+
+import org.apache.log4j.Logger;
+import org.wso2.security.tools.scanner.dependency.js.constants.JSScannerConstants;
+import org.wso2.security.tools.scanner.dependency.js.exception.ApiInvokerException;
+import org.wso2.security.tools.scanner.dependency.js.exception.ConfigParserException;
+import org.wso2.security.tools.scanner.dependency.js.exception.DownloaderException;
+import org.wso2.security.tools.scanner.dependency.js.model.Product;
+import org.wso2.security.tools.scanner.dependency.js.utils.CommonApiInvoker;
+import org.wso2.security.tools.scanner.dependency.js.utils.ConfigParser;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * This class is responsible to perform pre-processing tasks prior to execute retire.js scan.
+ * Pre processing includes following processes :
+ * Parse configuration
+ * Download product packs
+ * Download package.json files
+ * Unzip product directory
+ */
+public class PreProcessor {
+
+    private static final Logger log = Logger.getLogger(PreProcessor.class);
+    //Root directory where all products are downloaded.
+    private File productRootDirectory;
+    //Mapper to hold values each product's path where it is downloaded.
+    private HashMap<String, String> productFileMapper;
+
+    /**
+     * Entry point to start pre-processing task.
+     * Following tasks are sub task of this method.
+     * Parse configuration
+     * Download product packs
+     * Download package.json files
+     * Unzip product directory
+     *
+     * @return productFileMapper which holds the values each product's path where it is downloaded.
+     */
+    public HashMap<String, String> startPreprocessing() {
+
+        try {
+            List<String> productList;
+            //Parse config details.
+            //Get supported product list from config file
+            productList = ConfigParser.parseProductList();
+            List<Product> supportedProductList = new ArrayList<>();
+            //Get product details from respective product configuration files.
+            for (String product : productList) {
+                supportedProductList.add(ConfigParser.parseProductConfiguration(product));
+            }
+
+            //Create Root Directory for products if not exists
+            productRootDirectory = new File(JSScannerConstants.PRODUCT_HOME);
+            createDirectory(productRootDirectory);
+            log.info("[JS_SEC_DAILY_SCAN] Start downloading product packs");
+            //download resources
+            downloadResources(supportedProductList);
+            Arrays.fill(CommonApiInvoker.getGitToken(),
+                    JSScannerConstants.RANDOM_STRING.charAt(ConfigParser.getRandomNumber()));
+        } catch (ApiInvokerException | DownloaderException | ConfigParserException e) {
+            log.error("Error occurred while downloading product pack.", e);
+        }
+        return productFileMapper;
```

Comment:
From where is this populated? Should re-design the code. This code segment is hard to understand and maintain.